src/perf/bin/drvmain.cpp

```diff
@@ -219,11 +215,45 @@ SecNetPerfDriverUnload(
     CxPlatSystemUnload();
 }
 
+#define PERF_REG_PATH                 L"\\Parameters"
+
+_No_competing_thread_
+INITCODE
+NTSTATUS
+SecNetPerfGetServiceName(
+    _In_ PUNICODE_STRING BaseRegPath,
+    _Inout_ PUNICODE_STRING ServiceName
+    )
+{
+    USHORT BaseRegPathLength = BaseRegPath->Length / sizeof(WCHAR);
+    if (BaseRegPath->Buffer[BaseRegPathLength - 1] == L'\\') {
+        //LogWarning("[config] Trimming trailing '\\' from registry!");
+        BaseRegPathLength--;
+    }
+
+    //
+    // Get the service name from the base registry path.
+    //
+    USHORT ServiceNameLength = 0;
+    while (BaseRegPath->Buffer[BaseRegPathLength - ServiceNameLength - 1] != L'\\') {
+        ServiceNameLength++;
+    }
+    if (ServiceNameLength == 0) {
+        //LogError("[config] Empty service name!");
+        return STATUS_INVALID_PARAMETER;
+    }
+
+    UNICODE_STRING ServiceString = { ServiceNameLength, ServiceNameLength, BaseRegPath->Buffer + (BaseRegPathLength - ServiceNameLength) };
+
+    return RtlUnicodeStringCopy(ServiceName, &ServiceString);
```

Comment:
Do we really need to copy, or could we just reference the original memory?

Reply:
For this, yeah I could probably just reference the original memory.